gcp-observability/src/main/java/io/grpc/gcp/observability/interceptors/LogHelper.java

```diff
@@ -244,26 +239,21 @@ <T> void logRpcMessage(
       logger.log(Level.WARNING, "message is of UNKNOWN type, message and payload_size fields"
           + "of GrpcLogRecord proto will not be logged");
     }
-    PayloadBuilder<ByteString> pair = null;
+    PayloadBuilderHelper<Payload.Builder> pair = null;
     if (messageBytesArray != null) {
       pair = createMessageProto(messageBytesArray, maxMessageBytes);
     }
-
-    GrpcLogRecord.Builder logEntryBuilder = createTimestamp()
+    GrpcLogRecord.Builder logEntryBuilder = GrpcLogRecord.newBuilder()
         .setSequenceId(seqId)
         .setServiceName(serviceName)
         .setMethodName(methodName)
-        .setEventType(eventType)
-        .setEventLogger(eventLogger)
-        .setLogLevel(LogLevel.LOG_LEVEL_DEBUG)
-        .setRpcId(rpcId);
-    if (pair != null && pair.size != 0) {
-      logEntryBuilder.setPayloadSize(pair.size);
-    }
-    if (pair != null && pair.payload != null) {
-      logEntryBuilder.setMessage(pair.payload)
-          .setPayloadTruncated(pair.truncated);
-    }
+        .setAuthority(authority)
+        .setType(eventType)
+        .setLogger(eventLogger)
+        .setPayload(pair.payloadBuilder)
```

Comment:
`pair` can be null (line 243-245) in which case you will get NPE?

Reply:
Added a test to reproduce NPE. Added a null check before setting values of `GrpcLogRecord` using `pair`.